app/model/repository/ShadowAssignmentPointsRepository.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Model\Repository;
+
+use App\Model\Entity\ShadowAssignment;
+use App\Model\Entity\ShadowAssignmentPoints;
+use App\Model\Entity\User;
+use Kdyby\Doctrine\EntityManager;
+
+/**
+ * @method ShadowAssignmentPoints findOrThrow($id)
+ */
+class ShadowAssignmentPointsRepository extends BaseRepository {
+
+  public function __construct(EntityManager $em) {
+    parent::__construct($em, ShadowAssignmentPoints::class);
+  }
+
+  /**
+   * Find best solutions of given assignments for user.
+   * @param ShadowAssignment[] $shadowAssignments
+   * @param User $user
+   * @return ShadowAssignmentPoints[]
+   */
+  public function findEvaluationsForAssignment(array $shadowAssignments, User $user): array {
```

Comment:
Evaluations in the name -> should be points. And it does not make any sense, since Assignment is in singular here, yet an array of assignments is given.